src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -0,0 +1,343 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { SequelizeWorkspaceRepository } from './repositories/workspaces.repository';
+import { SequelizeUserRepository } from '../user/user.repository';
+import { UserUseCases } from '../user/user.usecase';
+import { MailerService } from '../../externals/mailer/mailer.service';
+import { ConfigService } from '@nestjs/config';
+import { createMock } from '@golevelup/ts-jest';
+import { WorkspacesUsecases } from './workspaces.usecase';
+import {
+  newUser,
+  newWorkspace,
+  newWorkspaceInvite,
+  newWorkspaceUser,
+} from '../../../test/fixtures';
+import { BadRequestException, NotFoundException } from '@nestjs/common';
+import { PreCreatedUser } from '../user/pre-created-user.domain';
+import { BridgeService } from '../../externals/bridge/bridge.service';
+
+jest.mock('../../middlewares/passport', () => {
+  const originalModule = jest.requireActual('../../middlewares/passport');
+  return {
+    __esModule: true,
+    ...originalModule,
+    Sign: jest.fn(() => 'newToken'),
+    SignEmail: jest.fn(() => 'token'),
+  };
+});
+
+describe('WorkspacesUsecases', () => {
+  let service: WorkspacesUsecases;
+  let workspaceRepository: SequelizeWorkspaceRepository;
+  let userRepository: SequelizeUserRepository;
+  let userUsecases: UserUseCases;
+  let mailerService: MailerService;
+  let networkService: BridgeService;
+  let configService: ConfigService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [WorkspacesUsecases],
+    })
+      .useMocker(() => createMock())
+      .compile();
+
+    service = module.get<WorkspacesUsecases>(WorkspacesUsecases);
+    workspaceRepository = module.get<SequelizeWorkspaceRepository>(
+      SequelizeWorkspaceRepository,
+    );
+    userRepository = module.get<SequelizeUserRepository>(
+      SequelizeUserRepository,
+    );
+    userUsecases = module.get<UserUseCases>(UserUseCases);
+    mailerService = module.get<MailerService>(MailerService);
+    networkService = module.get<BridgeService>(BridgeService);
+    configService = module.get<ConfigService>(ConfigService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('inviteUserToWorkspace', () => {
```

Comment:
I think there is a test case missing here: 
_When the invited user is not found, then it should throw_